public/app/features/dashboard/dashgrid/PanelStateWrapper.tsx

```diff
@@ -494,24 +495,33 @@ export class PanelStateWrapper extends PureComponent<Props, State> {
     return (
       <>
         <PanelContextProvider value={this.state.context}>
-          <PanelComponent
-            id={panel.id}
-            data={data}
-            title={panel.title}
-            timeRange={timeRange}
-            timeZone={this.props.dashboard.getTimezone()}
-            options={panelOptions}
-            fieldConfig={panel.fieldConfig}
-            transparent={panel.transparent}
-            width={innerWidth}
-            height={innerHeight}
-            renderCounter={renderCounter}
-            replaceVariables={panel.replaceVariables}
-            onOptionsChange={this.onOptionsChange}
-            onFieldConfigChange={this.onFieldConfigChange}
-            onChangeTimeRange={this.onChangeTimeRange}
-            eventBus={dashboard.events}
-          />
+          <PanelPerformanceMonitor
+            isInPanelEdit={dashboard.panelInEdit?.id === panel.id}
+            panelType={plugin.meta.id}
+            panelId={panel.id}
+            panelTitle={panel.title}
+            panelOptions={panelOptions}
+            panelFieldConfig={panel.fieldConfig}
+          >
```

Comment:
I wondering the value of nesting the panel under the performance component

might it be a little cleaner to just make `PanelPerformanceMonitor` and `PanelComponent` siblings underneath `PanelContextProvider`

unless this makes it easier to test? it seems we are re-rendering a fair bit to trigger functionality in jest 🤔

Reply:
This is a good point, I'll change them to be siblings. My initial logic was to have the monitoring component as a parent thinking it would help measuring panel paint times, but after testing it looks like measurements are virtually the same in both cases (component as parent and as sibling)

Reply:
yea, it feels like this should just be a call to log options when they change (and maybe on mount, but try to minimize duplicates log messages when opening the same dashboard over and over). and that function needs to do nothing when the logging is disabled 

the rendering performance check needs I think needs to be here somewhere in the render path, but also heavily sampled so we do not do it on every render.